prototypes/pipe-clamps.lua

```diff
@@ -161,6 +161,42 @@ local function build_picture_table(current_picture_table, current_name)
     }
 end
 
+local function build_composite_icon(current_entity)
+    local new_icons = {}
+
+    if current_entity.icons then
+        -- Icons table, transcribe layers into new_icons, fully define the IconData
+        for _, icon_data in pairs(current_entity.icons) do
+            table.insert(new_icons, {
+                icon = icon_data.icon,
+                icon_size = icon_data.icon_size or current_entity.icon_size,
+                icon_mipmaps = icon_data.icon_mipmaps or current_entity.icon_mipmaps,
+                tint = icon_data.tint,
+                shift = icon_data.shift,
+                scale = icon_data.scale or 1,
+            })
+        end
+    else
+        -- Standard icon, convert to an icons definition
+        table.insert(new_icons, {
+            icon = current_entity.icon,
+            icon_size = current_entity.icon_size,
+            icon_mipmaps = current_entity.icon_mipmaps,
+            scale = 1
+        })
+    end
+
+    -- Append the clamped_layer
+    table.insert(new_icons, {
+        icon = '__PickerPipeTools__/graphics/icons/lock.png',
```

Comment:
This table should be passed into the function via parameters to make the function generic.

Reply:
How general would you like this to be? I can generalize it so that it takes an IconData and inserts that into the `new_icons`, or it could take a table of IconData (so another `icons` definition)?

Reply:
Both would be acceptable. but IconData at the least
`If #passed_table > 1 then assume array of IconData`